src/main/java/ch/admin/bag/covidcertificate/sdk/core/verifier/nationalrules/DisplayValidityRangeCalculator.kt

```diff
@@ -35,18 +36,24 @@ internal class DisplayValidityRangeCalculator {
 		data: JsonNode,
 		certType: CertType
 	): ValidityRange? {
-		val displayFromDate: String = displayRules.find { it.id == "display-from-date" }?.logic ?: return null
-		val displayUntilDate: String = displayRules.find { it.id == "display-until-date" }?.logic ?: return null
-		val dateFromString = getValidity(displayFromDate, data, certType)
-		val dateUntilString = getValidity(displayUntilDate, data, certType)
+		val resultFromDate = evalRule(displayRules, "display-from-date", data) ?: return null
+		val resultUntilDate = evalRule(displayRules, "display-until-date", data) ?: return null
+		val dateFromString = getDateTime(resultFromDate, data, certType)
+		val dateUntilString = getDateTime(resultUntilDate, data, certType)
 		return ValidityRange(dateFromString, dateUntilString)
 	}
 
-	private fun getValidity(displayRule: String, data: JsonNode, certType: CertType): LocalDateTime? {
-		val displayLogic = jacksonMapper.readTree(displayRule)
-		val date = evaluate(displayLogic, data)
-		if (date is JsonDateTime) {
-			return getLocalDateTime(certType, date)
+	fun isOnlyValidInSwitzerland(
```

Comment:
Should this class maybe be renamed to DisplayValidityCalculator now? since it does more than only compute the ValidityRange

Reply:
done